controllers/civocluster_controller.go

```diff
@@ -103,6 +104,12 @@ func (r *CivoClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 // TODO Cluster deletion
 func (r *CivoClusterReconciler) delete(ctx context.Context, logger logr.Logger, civoCluster *infrastructurev1beta1.CivoCluster) (ctrl.Result, error) {
+	logger = log.FromContext(ctx)
+	logger.Info("Deleting civo cluster")
+	_, err := r.CivoClient.DeleteKubernetesCluster(*civoCluster.Spec.ID)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to delete cluster: %w", err)
+	}
```

Comment:
We also add finalizer to CivoCluster resource. So we need to remove it after actual cluster is deleted. You can do it with `controllerutil.RemoveFinalizer(civoCluster, civoClusterFinalizer)`.

Reply:
could i do something like :
```go
defer controllerutil.RemoveFinalizer(civoCluster, civoClusterFinalizer)
```

Reply:
I wouldn't do that. For example, what if `DeleteKubernetesCluster` returns error? Removing finalizer allows k8s to garbage collect this resource.

Reply:
Oh good point